src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/SvDiscoverFromLocalAssemblyContigAlignmentsSpark.java

```diff
@@ -306,21 +299,26 @@ private static void forNonComplexVariants(final EnumMap<RawTypes, JavaRDD<Assemb
 
     //==================================================================================================================
 
+    private static final SAMRecordQueryNameComparator samRecordQueryNameComparator = new SAMRecordQueryNameComparator();
+
     /**
      * write SAM file for provided {@code filteredContigs}
      * by extracting original alignments from {@code originalAlignments},
      * to directory specified by {@code outputDir}.
      */
     private static void writeSAM(final JavaRDD<AssemblyContigWithFineTunedAlignments> filteredContigs, final String rawTypeString,
                                  final JavaRDD<GATKRead> originalAlignments, final Broadcast<SAMFileHeader> headerBroadcast,
-                                 final String outputDir, final Logger toolLogger) {
+                                 final String outputPrefix, final Logger toolLogger) {
 
         final Set<String> filteredReadNames = new HashSet<>( filteredContigs.map(AssemblyContigWithFineTunedAlignments::getContigName).distinct().collect() );
         toolLogger.info(filteredReadNames.size() + " contigs indicating " + rawTypeString);
+        final SAMFileHeader header = headerBroadcast.getValue();
         final JavaRDD<SAMRecord> splitLongReads = originalAlignments.filter(read -> filteredReadNames.contains(read.getName()))
-                .map(read -> read.convertToSAMRecord(headerBroadcast.getValue()));
-        SVFileUtils.writeSAMFile(outputDir+"/"+rawTypeString+".sam", splitLongReads.collect().iterator(),
-                headerBroadcast.getValue(), false);
+                .map(read -> read.convertToSAMRecord(header));
+        header.setSortOrder(SAMFileHeader.SortOrder.queryname);
```

Comment:
Why sort these by queryname and not by position? I could be convinced either way but this requires an extra step to load the file into IGV.

Reply:
that was because I basically used Ribbon for reviewing those alignments, 
which is definitely easier with SAM compared to with BAM.
Now changed to adding an parameter to the utility 
* if BAM, sort by coordinate order.
* if SAM, sort by query name.